### DISCUSSION TITLE
Set Prediction Threshold

### DISCUSSION THREAD

**Discussion by krupeshp** [open-edge-platform/anomalib#2214] [Q&A]
Is there a way to set a prediction threshold for all the predictions so I can get a better mask? based out of a heatmap of the anomalies?
Here I have four anomalies but the heatmap shows only 3 and the mask shows 1.
![WhatsApp Image 2024-07-24 at 21 12 42_98e36db8](https://github.com/user-attachments/assets/0e0b201c-4576-4369-a238-bb7c7794380b)

**Comment by JinYuannn** [comment]
set the param `threshold` in `Engine`.

**Reply by krupeshp** [reply]
@samet-akcay @JinYuannn Can you give me an example please? threshold is F1AdaptiveScore or something else? Can i set it manual number like 0.3 ?

**Reply by samet-akcay** [reply]
By default `Engine` sets the threshold to adaptive threshold, where anomalib finds the best threshold that achieves the highest f1-score on the validation set. 
https://github.com/openvinotoolkit/anomalib/blob/d1f824a5798262891dbbe583fb291e1cf9aa7d2a/src/anomalib/engine/engine.py#L127

If you want to set a manual threshold, you could use `ManualThreshold` class 
https://github.com/openvinotoolkit/anomalib/blob/d1f824a5798262891dbbe583fb291e1cf9aa7d2a/src/anomalib/metrics/threshold/manual_threshold.py#L11-L47
